packages/dropdowns/src/styled/menu/StyledMenu.tsx

```diff
@@ -35,14 +40,16 @@ const retrieveMenuMargin = ({
   hasArrow?: boolean;
   placement?: POPPER_PLACEMENT;
 } & ThemeProps<DefaultTheme>) => {
-  const marginAmount = shouldShowArrow({ hasArrow, placement })
-    ? `${theme.space.base * 2}px`
-    : `${theme.space.base}px`;
-
   if (!placement) {
     return '';
   }
 
+  let marginAmount = shouldShowArrow({ hasArrow, placement })
+    ? getArrowSize({ theme })
+    : `${theme.space.base}px`;
+
+  marginAmount = math(`${marginAmount} - ${math(`${theme.borderWidths.sm} * 2`)}`);
```

Comment:
This is the wrong update, resulting in more problematic positioning than seen previously. We want the full 4px (or 8px for arrows) margin. It should not be reduced by border width.

The following pics try to highlight the 1px discrepancy seen before this code change. Usually these have something to do with unanticipated space being added to the DOM and my require display, alignment, or font-size tweaks in order to address...

![Screen Shot 2020-01-30 at 9 42 40 AM](https://user-images.githubusercontent.com/143773/73475670-a8156400-4345-11ea-9417-813d339ea206.png)

![Screen Shot 2020-01-30 at 9 42 16 AM](https://user-images.githubusercontent.com/143773/73475687-afd50880-4345-11ea-918f-413f58a1a091.png)

These, on the other hand, are displaying as desired:

![Screen Shot 2020-01-30 at 9 43 49 AM](https://user-images.githubusercontent.com/143773/73475721-bd8a8e00-4345-11ea-9b80-47637b19701c.png)

![Screen Shot 2020-01-30 at 9 44 23 AM](https://user-images.githubusercontent.com/143773/73475732-c2e7d880-4345-11ea-8441-88562fea9983.png)

Reply:
Actually, upon closer inspection that 3rd "end" placement above could use a tweak, too. It impedes over the button border by 1px.

Reply:
The 1px overlap for `start` and `end` positions seems to be a Popper calculation issue with the trigger. If I use a standard `<div>` it positions correctly.

While working to remove the `line-height` value I found that it is required for Popper to handle our `margin-*` animations correctly. Without a value the initial position of the popper is off-center.

I'm going to keep looking into this, but the positions seem to be valid now.

Reply:
It turns out any non-zero font-size also corrected this issue. I've now added a `font-size: 0.01px` rule with a styling comment.

The positioning should be accurate now!